galasa-maven-plugin/README.md

```diff
@@ -0,0 +1,198 @@
+# Maven Plugin for building Galasa
+
+This repository is an extension to Maven2 and includes goals that are used to build OSGI bundles and test catalogs for Galasa. The OSGI bundle repositories contain all the project information, configuration details and dependencies that are needed for building and running Galasa projects. The test catalog is used to manage Galasa test cases - attributes associated with test cases that are held in the catalog can used to schedule test runs.
+
+## How to build locally
+Use the `build-locally.sh` script to build this code locally.
+
+Environment variable over-rides:
+
+- `LOGS_DIR` - Optional. Where logs are placed. Defaults to creating a temporary directory.
+- `SOURCE_MAVEN` - Optional. Where a maven repository is from which the build will draw artifacts.
+- `DEBUG` - Optional. Defaults to 0 (off)
+- `GPG_PASSPHRASE` - Used to sign and verify artifacts during the build
+
+## How to use the plugin
+
+### Building a test catalog for a Java bundle
+
+This goal causes a test catalog to be constructed for all the tests in the child bundles of this maven project.
+
+Goal: `bundletestcat`
+
+Phase: `package`
+
+Input Parameters/Properties:
+- `galasa.skip.bundletestcatalog` required.
+
+Output:
+A test catalog file is generated holding references to all the test classes.
+
+Example:
+```
+<plugin>
+    <groupId>dev.galasa</groupId>
+    <artifactId>galasa-maven-plugin</artifactId>
+    <extensions>true</extensions>
+    <executions>
+        <execution>
+        <id>build-testcatalog</id>
+        <phase>package</phase>
+        <goals>
+            <goal>bundletestcat</goal>
+        </goals>
+        </execution>
+    </executions>
+</plugin>
+```
+
+### Building an OBR resource
+
+Input Parameters/Properties:
+- `galasa.obr.url.type` property = "obrUrlType" optional
+- `includeSelf` optional. Default value is `false`
+
+### Publishing a test catalog to the Galasa ecosystem/server
+
+Goal: `deploytestcat`
+
+Phase: `deploy`
+
+Input Parameters/Properties:
+- `galasa.test.stream` required. A string.
+- `galasa.token` optional. An access token for the galasa ecosystem, if that ecosytem is using authentication.
+- `galasa.bootstrap` required. A URL to the ecosystem.
+- `galasa.skip.bundletestcatalog` optional. A boolean. If set to true, the test catalog is not deployed to the Galasa server.
+- `galasa.skip.deploytestcatalog` optional. A boolean. If set to true, the test catalog is not deployed to the Galasa server.
```

Comment:
Unsure what the difference between these two are, I was meant to ask you last week. Can we discuss?

Reply:
Is the description meant to be different on these two lines?

Reply:
Explained more in the readme.